src/main/java/com/robinpowered/sdk/RobinApi.java

```diff
@@ -130,7 +130,7 @@ public void setCredential(Credential credential) {
      * @return Whether a credential has been set or not.
      */
     public boolean hasCredential() {
-        return credential != null;
+        return credential != null && credential.getValue() != null;
```

Comment:
This sort of confuses the interface, since it does have a credential. I think this change would break the interface. I think it might be better to leave as is and have the validity of the credential be validated by the client which is setting it.